crates/file_finder/src/file_finder_tests.rs

```diff
@@ -1061,6 +1061,70 @@ async fn test_search_sorts_history_items(cx: &mut gpui::TestAppContext) {
     });
 }
 
+#[gpui::test]
+async fn test_select_first_history_file_by_default(cx: &mut gpui::TestAppContext) {
+    let app_state = init_test(cx);
+
+    app_state
+        .fs
+        .as_fake()
+        .insert_tree(
+            "/root",
+            json!({
+                "test": {
+                    "1_qw": "// First file that matches the query",
+                    "2_second": "// Second file",
+                    "3_third": "// Third file",
+                }
+            }),
+        )
+        .await;
+
+    let project = Project::test(app_state.fs.clone(), ["/root".as_ref()], cx).await;
+    let (workspace, cx) = cx.add_window_view(|cx| Workspace::test_new(project, cx));
+    // generate some history to select from, but leave the last buffer open
+    open_close_queried_buffer("1", 1, "1_qw", &workspace, cx).await;
+    open_close_queried_buffer("2", 1, "2_second", &workspace, cx).await;
+    open_queried_buffer("3", 1, "3_third", &workspace, cx).await;
+
+    let pricker = open_file_picker(&workspace, cx);
+    pricker.update(cx, |finder, _| {
+        assert_eq!(finder.delegate.selected_index(), 1);
```

Comment:
To be on the safe side, let's assert the delegate contents too: to prove that the selected index is on the right file path.

Reply:
I've added checks for the selected file names in addition to indexes